messenger/messenger.go

```diff
@@ -214,8 +198,10 @@ func (m *MesosMessenger) Send(ctx context.Context, upid *upid.UPID, msg proto.Me
 // 1) routing internal error to callback handlers
 // 2) testing components without starting remote servers.
 func (m *MesosMessenger) Route(ctx context.Context, upid *upid.UPID, msg proto.Message) error {
-	// if destination is not self, send to outbound.
-	if !upid.Equal(m.upid) {
+	if upid == nil {
+		panic("cannot route a message to a nil pid")
```

Comment:
Because I had to draw the line somewhere. There's still lots of places in
the code base where UPID pointers are passed around. The initial goal was
to eliminate the sharing between the transport and messenger in the case
where the "self" UPID could change in an unguarded (non-critical section)
context.

On Thu, Sep 24, 2015 at 5:49 AM, Tomás Senart <notifications@github.com>
wrote:

> In messenger/messenger.go
> <https://github.com/mesos/mesos-go/pull/189#discussion_r40302025>:
>
> > @@ -214,8 +198,10 @@ func (m *MesosMessenger) Send(ctx context.Context, upid *upid.UPID, msg proto.Me
> >  // 1) routing internal error to callback handlers
> >  // 2) testing components without starting remote servers.
> >  func (m *MesosMessenger) Route(ctx context.Context, upid *upid.UPID, msg proto.Message) error {
> > -	// if destination is not self, send to outbound.
> > -	if !upid.Equal(m.upid) {
> > +	if upid == nil {
> > +		panic("cannot route a message to a nil pid")
>
> Why don't we change the signature of the method to not take a pointer
> instead?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mesos/mesos-go/pull/189/files#r40302025>.
>